### PR TITLE
Adds support for specifying a quality indexed array of URLs

### PR DESCRIPTION
**Pull request by garrettmoon**
This allows PINRemoteImage to download from a set of URLs depending on network conditions.

More context perhaps:
https://github.com/pinterest/PINRemoteImage/pull/407
https://github.com/pinterest/PINRemoteImage/pull/403

### CODE REVIEW

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any source code changes should have an entry in CHANGELOG.md or have #trivial in their title.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/9285d2a5422868c914bb47f64c1e4b88/raw/1ee240aea84a3fa8c126346aae548149c62feb37/log.txt)

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any source code changes should have an entry in CHANGELOG.md or have #trivial in their title.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/db093160ab1157b0a2ef2fe919653387/raw/6799a38c0ee27930415f5108fabb97e7390023fb/log.txt)

File: Source/ASNetworkImageNode.mm [hunk 1]
```diff
@@ -322,13 +349,15 @@ - (void)displayWillStartAsynchronously:(BOOL)asynchronously
...
-        _imageLoaded = YES;
+    if (_imageLoaded == NO && _URLs.count > 0 && _downloadIdentifier == nil) {
+      for (NSURL *url in _URLs) {
+        UIImage *result = [[_cache synchronouslyFetchedCachedImageWithURL:url] asdk_image];
+        if (result) {
+          [self _locked_setCurrentImageQuality:1.0];
+          [self _locked__setImage:result];
+          
+          _imageLoaded = YES;
+        }
```
Comment (nguyenhuy) [inline Source/ASNetworkImageNode.mm]: Looks like we're fetching images from low cost to higher cost and don't bail as soon as there is a cache hit. Does this mean that if there are multiple images, we'll end up fetching all of them but only use the best one? If that's the case, should we instead fetch in reverse order and stop as early as possible?

Comment (garrettmoon) [reply Source/ASNetworkImageNode.mm]: Yeah, good catch!

**Pull request merged**